### PR TITLE
devguide: add page on Frame support - v6

### PR DESCRIPTION
**Pull request by jufajardini**
Previous PR: https://github.com/OISF/suricata/pull/6933

Describe changes:
- improve explanation about `frame_len`, with regards to frame registering

TODO:
- once the page about the frame keyword has been merged, update this one to link to it

Link to redmine ticket:
https://redmine.openinfosecfoundation.org/issues/4697

Compiled version can be seen at
https://stream-frame-support-v6.surge.sh/extending/app-layer/app-layer-frames.html

### CODE REVIEW

File: doc/devguide/extending/app-layer/app-layer-frames.rst [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+The goal of the stream *Frame* is to expose application layer protocol `PDUs <https://en.wikipedia.org/wiki/Protocol_data_unit>`_ and other such arbitrary elements to the detection engine directly, instead of relying on Transactions. The main purpose is to bring *TCP.data* processing times down by specialising/ filtering down traffic detection.
+
+Adding Frame Support to a Parser
+================================
+
+The application layer parser exposes frames it supports to the detect engine, by tagging them as they're parsed. The rest works automatically.
+
+In order to allow the engine to identify frames for records of a given application layer parser, you'll have to decide which frames make sense for the specific protocol you are handling. Once you have that, the basic steps are:
+
+- create an enum for the frame types;
```
Comment (victorjulien) [inline doc/devguide/extending/app-layer/app-layer-frames.rst:36]: also register relevant callbacks, eg.
```
        get_frame_id_by_name: Some(TelnetFrameType::ffi_id_from_name),
        get_frame_name_by_id: Some(TelnetFrameType::ffi_name_from_id),
```

File: doc/devguide/extending/app-layer/app-layer-frames.rst [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+And to the client:
+
+.. literalinclude:: ../../../../rust/src/frames.rs
+    :caption: rust/src/frames.rs
+    :language: rust
+    :start-at: pub fn new_tc
+    :lines: 1-6
+    :dedent: 4
+
```
Comment (jufajardini) [inline doc/devguide/extending/app-layer/app-layer-frames.rst:104]: remove these

File: doc/devguide/extending/app-layer/app-layer-frames.rst [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+The application layer parser exposes frames it supports to the detect engine, by tagging them as they're parsed. The rest works automatically.
+
+In order to allow the engine to identify frames for records of a given application layer parser, you'll have to decide which frames make sense for the specific protocol you are handling. Once you have that, the basic steps are:
+
+- create an enum for the frame types;
+- identify the parsing function(s) where application layer records are parsed;
+- identify the correct moment to use the calls;
+- use the Frame API calls directly or build upon them to register the frames.
+
+Once this is done, you can enable frame eve-output to confirm that your frames are being registered correctly. If done correctly, you should be able to write a rule using the *frame* keyword and the frame type you have registered with the application layer parser.
```
Comment (jufajardini) [inline doc/devguide/extending/app-layer/app-layer-frames.rst:41]: rewrite, remove repetition.

Comment (jufajardini) [reply doc/devguide/extending/app-layer/app-layer-frames.rst:41]: mention that there's a hard coded limit to printable payload.

**Comment by jufajardini** [comment]
Feedback incorporated on: https://github.com/OISF/suricata/pull/6946

**Pull request closed without merging**